Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/CustomFilePickerFragment.java

```diff
@@ -41,4 +44,33 @@ public Uri toUri(@NonNull final File file)
       cancel.setVisibility(View.GONE);
     }
   }
+
+  @Override
+  protected boolean isItemVisible(final File file)
+  {
+    // Some users jump to the conclusion that Dolphin isn't able to detect their
+    // files if the files don't show up in the file picker when mode == MODE_DIR.
+    // To avoid this, show files even when the user needs to select a directory.
+
+    return (showHiddenItems || !file.isHidden()) &&
+            (file.isDirectory() ||
+                    extensions.contains(fileExtension(file.getName()).toLowerCase()));
+  }
+
+  @Override
+  public boolean isCheckable(@NonNull final File file)
+  {
+    // We need to make a small correction to the isCheckable logic due to
+    // overriding isItemVisible to show files when mode == MODE_DIR.
+    // AbstractFilePickerFragment always treats files as checkable when
+    // allowExistingFile == true, but we don't want files to be checkable when mode == MODE_DIR.
+
+    return super.isCheckable(file) && !(mode == MODE_DIR && file.isFile());
+  }
+
+  private String fileExtension(String filename)
```

Comment:
This could probably be `static`. Also, the lack of a `File.getExtension()` method in Java is disturbing...